types/next/index.d.ts

```diff
@@ -1,16 +1,29 @@
-// Type definitions for next 2.4
+// Type definitions for next 5.0
 // Project: https://github.com/zeit/next.js
 // Definitions by: Drew Hays <https://github.com/dru89>
 //                 Brice BERNARD <https://github.com/brikou>
+//                 James Hegedus <https://github.com/jthegedus>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.6
 
 /// <reference types="node" />
 
 import * as http from "http";
 import * as url from "url";
+import * as fetch from "isomorphic-unfetch";
```

Comment:
Not sure how to do this since the `jsonPageRes` key has type of a [Fetch Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object (client only)

Reply:
See the `///` directive a few lines up that allows the `http` and `url` types to be used within this declaration file.

https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html